### ISSUE TITLE
Errors while running Translate.py sample code with multiple GPUS

### ISSUE THREAD

**Issue by suraj1990** [tensorflow/tensorflow#2944]
GitHub issues are for bugs / installation problems / feature requests.  
For general support from the community, see [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow).
To make bugs and feature requests more easy to find and organize, we close issues that are deemed
out of scope for GitHub Issues and point people to StackOverflow.

For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.

### Environment info
Operating System: Ubuntu

Installed version of CUDA and cuDNN: 7.5 and 7
(please attach the output of `ls -l /path/to/cuda/lib/libcud*`):

If installed from binary pip package, provide:

1. Which pip package you installed.
2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`. 0.8

If installed from sources, provide the commit hash:

### Steps to reproduce
1. running the translate.py example
2.Crashes while creating models after tokenizing the training data.

The sample code seems to run fine with single GPU of 4 GB memory. I am using AWS GPU instances. But it works fine for basic values, ie vocab size of 40000 and size of 512. I want to train on a vocab size of 500000 and size of 1024. This is the reason why I opted to go for 4 GPUs now. 
(In AWS terms, g2.2xlarge to g2.8xlarge). 
![g2-8xlarge-crash](https://cloud.githubusercontent.com/assets/5551707/16170060/c599bef8-3562-11e6-9723-4a38999c6a43.png)
![nvidia-smi-details](https://cloud.githubusercontent.com/assets/5551707/16170062/d7a121fe-3562-11e6-928e-3270ebaa44b9.png)

### Logs or other output that would be helpful
(If logs are large, please upload as attachment).

**Comment by suraj1990** [comment]
@lukaszkaiser Can you give me any inputs?

**Comment by prb12** [comment]
In order to understand what is happening here, you will at least need to provide the command lines you are using and the full output of the commands which are failing.  (e.g. There is not enough information in the two screenshots above)

@lukaszkaiser  Are there any known issues with running this model multi-GPU?

**Comment by lukaszkaiser** [comment]
I don't know of any issues. Could you provide a text log with full error message? (The screens don't seem to include it all, a text file with the whole error message would be more helpful.)

**Comment by suraj1990** [comment]
[logs.txt](https://github.com/tensorflow/tensorflow/files/340736/logs.txt)
Sorry for reporting this issue, I feel this is not a bug but a problem related to my system memory.
I tried running with varied size parameter. it runs for size 256 and 512 but for 1024 there is a crash. I am giving 22M sentences, and vocab size of 1M.  When I checked the logs for 1024, I saw that the crash is because of unavailability of memory. This is on a AWS instance with 4 GPU's of 4Gb each. I have attached the logs.
